hrm-service/src/case/case-sql.ts

```diff
@@ -94,59 +100,61 @@ const SEARCH_WHERE_CLAUSE = `WHERE
-          "contacts.\"rawJson\"->'childInformation'->'name'->>'firstName'",
-          "contacts.\"rawJson\"->'childInformation'->'name'->>'lastName'",
-          [
-            "contacts.\"rawJson\"->'childInformation'->'location'->>'phone1'",
-            "contacts.\"rawJson\"->'childInformation'->'location'->>'phone2'",
-            'contacts.number',
-          ],
-        )}  
+      ($<firstName> IS NULL AND $<lastName> IS NULL AND $<phoneNumber> IS NULL) OR
+      EXISTS (
+        SELECT 1 FROM "Contacts" c WHERE c."caseId" = cases.id 
+          AND (
+            (
+            ${nameAndPhoneNumberSearchSql(
+              "c.\"rawJson\"->'childInformation'->'name'->>'firstName'",
+              "c.\"rawJson\"->'childInformation'->'name'->>'lastName'",
+              [
+                "c.\"rawJson\"->'childInformation'->'location'->>'phone1'",
+                "c.\"rawJson\"->'childInformation'->'location'->>'phone2'",
+                'c.number',
+              ],
+            )})
+            -- search on callerInformation of connectedContacts
+            OR ( 
+              ${nameAndPhoneNumberSearchSql(
+                "c.\"rawJson\"->'callerInformation'->'name'->>'firstName'",
+                "c.\"rawJson\"->'callerInformation'->'name'->>'lastName'",
+                [
+                  "c.\"rawJson\"->'callerInformation'->'location'->>'phone1'",
+                  "c.\"rawJson\"->'callerInformation'->'location'->>'phone2'",
+                ],
+              )}  
+            )
+          )
       )
-        -- search on callerInformation of connectedContacts
-      OR ( 
-        ${nameAndPhoneNumberSearchSql(
-          "contacts.\"rawJson\"->'callerInformation'->'name'->>'firstName'",
-          "contacts.\"rawJson\"->'callerInformation'->'name'->>'lastName'",
-          [
-            "contacts.\"rawJson\"->'callerInformation'->'location'->>'phone1'",
-            "contacts.\"rawJson\"->'callerInformation'->'location'->>'phone2'",
-            'contacts.number',
-          ],
-        )}  
-      )
-        -- search on case sections in the expected format
-      OR (
-        EXISTS (
+      -- search on case sections in the expected format
+      OR EXISTS (
         SELECT 1 FROM "CaseSections" cs WHERE cs."caseId" = cases.id
-        AND
-        ${nameAndPhoneNumberSearchSql(
-          'cs."sectionTypeSpecificData"->>\'firstName\'',
-          'cs."sectionTypeSpecificData"->>\'lastName\'',
-          [
-            'cs."sectionTypeSpecificData"->>\'phone1\'',
-            'cs."sectionTypeSpecificData"->>\'phone2\'',
-          ],
-        )}
-        )
+          AND
+          ${nameAndPhoneNumberSearchSql(
+            'cs."sectionTypeSpecificData"->>\'firstName\'',
+            'cs."sectionTypeSpecificData"->>\'lastName\'',
+            [
+              'cs."sectionTypeSpecificData"->>\'phone1\'',
+              'cs."sectionTypeSpecificData"->>\'phone2\'',
+            ],
+          )}
```

Comment:
Given the flexibility we are giving to case sections in general, I wonder if it's better to continue "constraining" the helplines to just use this 4 properties to search on, or if we should just search over any property inside of `sectionTypeSpecificData`.